pkg/isb/stores/jetstream/reader.go

```diff
@@ -119,12 +119,27 @@ func (jr *jetStreamReader) Read(_ context.Context, count int64) ([]*isb.ReadMess
 	}(time.Now())
 	var err error
 	var result []*isb.ReadMessage
-	msgs, err := jr.sub.Fetch(int(count), nats.MaxWait(jr.opts.readTimeOut))
+	// We use FetchBatch to have batching in our code with two parameters, ie ReadBatchSize
+	// and ReadTimeout. This function should read from Jetstream until we have a read ReadBatchSize
+	// number of messages or the ReadTimeout duration has elapsed.
+	msgs, err := jr.sub.FetchBatch(int(count), nats.MaxWait(jr.opts.readTimeOut))
 	if err != nil && !errors.Is(err, nats.ErrTimeout) {
 		isbReadErrors.With(map[string]string{"buffer": jr.GetName()}).Inc()
 		return nil, fmt.Errorf("failed to fetch messages from jet stream subject %q, %w", jr.subject, err)
 	}
-	for _, msg := range msgs {
+
+	// Wait until the batch processing is done
+	<-msgs.Done()
```

Comment:
Be very cautious on this async fetch call, I don't trust the reliability based on our past experience.

Reply:
@kohlisid, can you check whether it uses `Fetch` behind the scenes?

Reply:
It does not use Fetch call directly. Both of them have different ways of collating and reporting the data. 
@whynowy Any specific areas that we can verify to ensure the scenarios you are mentioning?

Reply:
Additionally, if you feel there is high risk. We can use this method only when Batching is requested by the user, and the default could be `Fetch`